clang/include/clang/3C/TypeVariableAnalysis.h

```diff
@@ -58,6 +62,12 @@ class TypeVariableEntry {
   // A single constraint variable for solving the checked type of the type
   // variable. It is constrained GEQ all elements of ArgConsVars.
   ConstraintVariable *TypeParamConsVar;
+
+  // If an argument is a single identifier, store the constraint variable
+  // to recognize changes in type from inferred generics. Null otherwise.
+  // Meaningless if `TypeParamConsVar` has a basetype other than void, and
+  // when we have generic index constraints, those should be favored over this
+  ConstraintVariable *IdentArgumentCV;
```

Comment:
I don't follow this explanation. What's the purpose of the new field?

Reply:
Since we don't have generic index propagation, we need a workaround here. `TypeParamConsVar` above is needed for wildness constraints, but it isn't enough, because the generic could be added or cancelled between the time this code runs and the results are rewritten. `IdentArgumentCV` stores the CV of the actual argument to the function in order to check its generic index later. But we shouldn't use it as wildness constraint because it's a workaround and should be removed when generic propagation is implemented.

I'll make the names more consistent.